File: v1/src/main/java/com/api/v1/car/controllers/AllCarsDeletionController.java
package com.api.v1.car.controllers;

import com.api.v1.car.annotations.VIN;
import com.api.v1.car.exceptions.CarEntityNotExistException;
import com.api.v1.car.services.AllCarsDeletionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/cars")
public class AllCarsDeletionController {

    @Autowired
    private AllCarsDeletionService service;

    @DeleteMapping
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    public Mono<Void> deleteAll() {
        return service.deleteAll();
    }

    @DeleteMapping("{vin}")
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    public Mono<Void> deleteByVin(@VIN @PathVariable String vin) {
        return service.deleteByVin(vin);
    }

}

File: v2/src/main/kotlin/com/api/v2/car/controllers/AllCarsDeletionController.kt
package com.api.v2.car.controllers

import com.api.v2.car.services.AllCarsDeletionService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v1/cars")
class AllCarsDeletionController {

    @Autowired
    private lateinit var service: AllCarsDeletionService

    @DeleteMapping
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    suspend fun deleteAll() {
        return service.deleteAll()
    }

    @DeleteMapping("{vin}")
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    suspend fun deleteByVin(@PathVariable vin: String) {
        return service.deleteByVin(vin)
    }

}